projects/plugins/jetpack/modules/comments/subscription-modal-on-comment/class-jetpack-subscription-modal-on-comment.php

```diff
@@ -0,0 +1,224 @@
+	 * This can then be edited by the user.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_template_content() {
+		$discover_more_from = __( 'Never miss a beat!', 'default' ); // phpcs:ignore WordPress.WP.I18n.TextDomainMismatch
+		$subscribe_text     = __( 'Interested in getting blog post updates? Simply click the button below to stay in the loop!', 'default' ); // phpcs:ignore WordPress.WP.I18n.TextDomainMismatch
+		$subscribe_button   = __( 'Subscribe', 'default' ); // phpcs:ignore WordPress.WP.I18n.TextDomainMismatch
+
+		return <<<HTML
+	<!-- wp:group -->
+	<div class="jetpack-subscription-modal__iframe-container">
+		<iframe
+			class="jetpack-subscription-modal__iframe"
+			frameBorder="0"
+			allowTransparency="1"
+			src="about:blank"
+			id="jetpack-subscription-modal__iframe"
+		></iframe>
+	</div>
+	<!-- /wp:group -->
+	<!-- wp:group {"style":{"spacing":{"top":"32px","bottom":"32px","left":"32px","right":"32px"},"margin":{"top":"0","bottom":"0"}},"border":{"color":"#dddddd","width":"1px"}},"layout":{"type":"constrained","contentSize":"450px"}} -->
+	<div class="wp-block-group has-border-color jetpack-subscription-modal__modal-content-form" style="border-color:#dddddd;border-width:1px;margin-top:0;margin-bottom:0;padding-top:0;padding-right:32px;padding-bottom:32px;padding-left:32px">
+
+		<!-- wp:paragraph {"align":"right","style":{"spacing":{"margin":{"top":"0"}},"typography":{"fontSize":"14px"}},"className":"jetpack-subscription-modal__close"} -->
+		<p class="has-text-align-center jetpack-subscription-modal__close" style="margin-top:20px;font-size:14px;text-align:right"><a href="#"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" aria-hidden="true" width="24" height="24" focusable="false"><path d="M12 13.06l3.712 3.713 1.061-1.06L13.061 12l3.712-3.712-1.06-1.06L12 10.938 8.288 7.227l-1.061 1.06L10.939 12l-3.712 3.712 1.06 1.061L12 13.061z"></path></svg></a></p>
+		<!-- /wp:paragraph -->
+
+		<!-- wp:heading {"textAlign":"center","style":{"typography":{"fontStyle":"normal","fontWeight":"600","fontSize":"26px"},"layout":{"selfStretch":"fit","flexSize":null},"spacing":{"margin":{"top":"4px","bottom":"10px"}}}} -->
+		<h2 class="wp-block-heading has-text-align-center" style="margin-top:4px;margin-bottom:10px;font-size:26px;font-style:normal;font-weight:600">$discover_more_from</h2>
+		<!-- /wp:heading -->
+
+		<!-- wp:paragraph {"align":"center","style":{"typography":{"fontSize":"15px"},"spacing":{"margin":{"top":"4px","bottom":"0px"}}}} -->
+		<p class='has-text-align-center' style='margin-top:4px;margin-bottom:0px;font-size:15px'>$subscribe_text</p>
+		<!-- /wp:paragraph -->
+
+		<!-- wp:group {"style":{"spacing":{"top":"32px","bottom":"32px","left":"32px","right":"32px"},"margin":{"top":"0","bottom":"0"}},"border":{"color":"#dddddd","width":"1px"}},"layout":{"type":"constrained","contentSize":"450px"}} -->
+		<form class="jetpack-subscription-modal__form">
```

Comment:
Why not just render subscribe block directly here? 

You're replicating a lot of styles and functionality that you'd get for free.

You could extend block's parameters to allow passing pre-defined email.

Reply:
That would also allow the block to inherit theme styles directly, so it'll match everything else on the site better.

Reply:
Example:

https://github.com/Automattic/jetpack/blob/e849efbc01ccb083925969dd38994c80b242a843/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L1058

Reply:
I worked on the custom input since we need three features not included in the block:
- Predefined email
- Read only input (if user is logged-in)
- Custom param (app_source) for tracking purpose

Do you think we can extend the block to use this features instead of what we did?

Reply:
Totally! :-)

We're already adding the custom tracking source here: https://github.com/Automattic/jetpack/pull/34792 (help testing welcome!). It works with D133298-code.

I think it would be good to add another attribute which passes the email address to the block.

"Button only" style could also be helpful, I could finish this next week or you can also take over: https://github.com/Automattic/jetpack/pull/33921

Reply:
That's interesting, thanks 🙇 

Since I didn't know the user's email nor was logged in until I received the comment response, I can update the email address and make the input read-only using javascript like here: https://github.com/Automattic/jetpack/pull/34659/files#diff-3381b2180b188f8bfc6654da80950e0057d5f34b7b4b02eef11f216d33f56360R86

I think we can make it work using only this PR: https://github.com/Automattic/jetpack/pull/34792